tests/components/smart_meter_texas/conftest.py

```diff
@@ -0,0 +1,80 @@
+"""Test configuration and mocks for Smart Meter Texas."""
+import asyncio
+import json
+from pathlib import Path
+
+import pytest
+from smart_meter_texas.const import (
+    AUTH_ENDPOINT,
+    BASE_ENDPOINT,
+    BASE_URL,
+    LATEST_OD_READ_ENDPOINT,
+    METER_ENDPOINT,
+    OD_READ_ENDPOINT,
+)
+
+from homeassistant.components.smart_meter_texas.const import DATA_COORDINATOR, DOMAIN
+
+from tests.common import MockConfigEntry, load_fixture
+
+
+def load_smt_fixture(name):
+    """Return a dict of the json fixture."""
+    json_fixture = load_fixture(Path() / DOMAIN / f"{name}.json")
+    return json.loads(json_fixture)
+
+
+async def setup_integration(hass, config_entry, aioclient_mock):
+    """Initialize the Smart Meter Texas integration for testing."""
+    mock_connection(aioclient_mock)
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+
+async def refresh_data(hass, config_entry, aioclient_mock):
+    """Mock a DataUpdateCoordinator async_refresh."""
+    mock_connection(aioclient_mock)
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
+    await coordinator.async_refresh()
+    await hass.async_block_till_done()
```

Comment:
We should call the generic entity refresh service here instead of accessing `hass.data` directly.  `homeassistant.update_entity`

Reply:
I think this one got missed

Reply:
Whoops. Sorry about that.